test/CardPaymentProcessorUpgradeable.test.ts

```diff
@@ -0,0 +1,1461 @@
+  const REVERT_MESSAGE_IF_TOKEN_TRANSFER_AMOUNT_EXCEEDS_BALANCE =
+    "ERC20: transfer amount exceeds balance";
+  const REVERT_MESSAGE_IF_PAYMENT_AUTHORIZATION_ID_ALREADY_EXISTS =
+    "CardPaymentProcessor: payment with the provided authorization ID already exists and was not revoked";
+  const REVERT_MESSAGE_IF_PAYMENT_DOES_NOT_EXIST =
+    "CardPaymentProcessor: payment with the provided authorization ID does not exist";
+  const REVERT_MESSAGE_IF_PAYMENT_IS_CLEARED =
+    "CardPaymentProcessor: payment with the provided authorization ID is cleared";
+  const REVERT_MESSAGE_IF_PAYMENT_IS_UNCLEARED =
+    "CardPaymentProcessor: payment with the provided authorization ID is uncleared";
+  const REVERT_MESSAGE_IF_PARENT_TX_HASH_IS_ZERO =
+    "CardPaymentProcessor: parent transaction hash should not equal 0";
+  const REVERT_MESSAGE_IF_PAYMENT_HAS_INAPPROPRIATE_STATUS =
+    "CardPaymentProcessor: payment with the provided authorization ID has an inappropriate status";
+  const REVERT_MESSAGE_IF_PAYMENT_REVOCATION_COUNTER_HAS_REACHED_MAXIMUM =
+    "CardPaymentProcessor: revocation counter of the payment has reached the configured maximum";
+
+  let cardPaymentProcessor: Contract;
+  let tokenMock: Contract;
+  let deployer: SignerWithAddress;
+  let user1: SignerWithAddress;
+  let user2: SignerWithAddress;
+  let ownerRole: string;
+  let pauserRole: string;
+  let executorRole: string;
+
+  async function setUpContractsForPayments(payments: TestPayment[]) {
+    for (let payment of payments) {
+      await proveTx(tokenMock.mint(payment.account.address, payment.amount));
+      const allowance: BigNumber = await tokenMock.allowance(payment.account.address, cardPaymentProcessor.address);
+      if (allowance.lt(BigNumber.from(ethers.constants.MaxUint256))) {
+        await proveTx(tokenMock.connect(payment.account).approve(
+          cardPaymentProcessor.address,
+          ethers.constants.MaxUint256
+        ));
+      }
+    }
+  }
+
+  async function makePayments(payments: TestPayment[]) {
+    for (let payment of payments) {
+      await proveTx(cardPaymentProcessor.connect(payment.account).makePayment(
+        payment.amount,
+        createBytesString(payment.authorizationId, BYTES16_LENGTH),
+        createBytesString(payment.makingPaymentCorrelationId, BYTES16_LENGTH)
+      ));
+      payment.status = PaymentStatus.Uncleared;
+    }
+  }
+
+  async function setExecutorRole(account: SignerWithAddress) {
+    await proveTx(cardPaymentProcessor.grantRole(executorRole, account.address));
+  }
+
+  async function clearPayments(payments: TestPayment[], adminAccount: SignerWithAddress) {
+    const authorizationIds: string[] = [];
+    payments.forEach((payment: TestPayment) => {
+      authorizationIds.push(createBytesString(payment.authorizationId, BYTES16_LENGTH));
+      payment.status = PaymentStatus.Cleared;
+    });
+    await proveTx(cardPaymentProcessor.connect(adminAccount).clearPayments(authorizationIds));
```

Comment:
Codacy has a fix for the issue: [Replace `cardPaymentProcessor.connect(adminAccount).clearPayments(authorizationIds)` with `⏎······cardPaymentProcessor.connect(adminAccount).clearPayments(authorizationIds)⏎····`](https://app.codacy.com/gh/cloudwalk/brlc-periphery/pullRequest?prid=9942053)
```suggestion
    await proveTx(
```